src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,61 +1,71 @@
 import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import { isTwoDimensionsData } from "../../../helpers/executionResultHelper";
+import PushpinSizeLegend, { IPushpinSizeLegendProps } from "./legends/PushpinSizeLegend";
+import PushpinCategoryLegend, {
+    IPushpinCategoryLegendProps,
+} from "../geoChart/legends/PushpinCategoryLegend";
 
 export interface IGeoChartLegendRendererProps {
     config: IGeoConfig;
-    execution: Execution.IExecutionResponses;
+    geoData: IGeoData;
+    sizeData?: number[];
+    segmentData?: IPushpinCategoryLegendItem[];
+    sizeFormat?: string;
     locale: string;
     position?: string;
+    onItemClick?(item: IPushpinCategoryLegendItem): void;
 }
+
 export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    if (!props.execution) {
+    const { position = TOP, geoData } = props;
+    const { size, segment } = geoData;
+    const isLegendHidden = !size && !segment;
+    if (isLegendHidden) {
         return null;
     }
-    const {
-        execution: { executionResult, executionResponse },
-        config: { mdObject: { buckets = [] } = {} },
-        locale,
-        position = TOP,
-    } = props;
-    const geoData: IGeoData = getGeoData(buckets, executionResponse.dimensions);
-    const { data } = executionResult;
-    const { size } = geoData;
     const classes = `geo-legend s-geo-legend position-${position}`;
-
-    if (!size || !isTwoDimensionsData(data)) {
-        return null;
-    }
     return (
         <div className={classes}>
-            {size &&
-                renderPushpinSizeLegend(
-                    data[size.index].map(stringToFloat),
-                    getFormatFromExecutionResponse(size.index, executionResponse),
-                    locale,
-                )}
+            {segment && renderPushpinCategoryLegend(props)}
+            {size && renderPushpinSizeLegend(props)}
         </div>
     );
 }
 
-function renderPushpinSizeLegend(sizeValues: number[], format: string, locale: string): JSX.Element {
-    const values: number[] = sizeValues.filter(isFinite);
+function renderPushpinSizeLegend(props: IGeoChartLegendRendererProps): JSX.Element {
+    const { sizeData = [], sizeFormat, geoData, locale } = props;
+    const sizes: number[] = sizeData.filter(isFinite);
+    const { size } = geoData;
+
     return (
         <IntlWrapper locale={locale}>
             <IntlTranslationsProvider>
-                {(props: ITranslationsComponentProps) => (
-                    <PushpinSizeLegend numericSymbols={props.numericSymbols} format={format} sizes={values} />
-                )}
+                {(props: ITranslationsComponentProps) => {
+                    const legendProps: IPushpinSizeLegendProps = {
+                        sizes,
+                        format: sizeFormat,
+                        numericSymbols: props.numericSymbols,
+                        measureName: size.name,
```

Comment:
It should be:
const { sizeData = [], sizeFormat, geoData, locale, numericSymbols } = props;
const sizes: number[] = sizeData.filter(isFinite);
const { size: { name } = geoData;

Reply:
numericSymbols don't belong IGeoChartLegendRendererProps.  It's a part a ITranslationsComponentProps